### ISSUE TITLE
V2 panics using Svelte template

### ISSUE THREAD

**Issue by BillBuilt** [wailsapp/wails#1055]
Hello, having some issues trying to use the Svelte template. Thanks in advance and I really love what you are doing!

**Description**
When launching `npm run dev` in a second  terminal as per the directions at `https://github.com/raitonoberu/wails-svelte-template`, Wails panics as follows:

```
panic: Unexpected host for request on wails:// scheme
goroutine 7 [running]:
github.com/wailsapp/wails/v2/internal/frontend/desktop/darwin.(*Frontend).processRequest(0xc00019e840, 0xc0001e6210)
	/Users/billmatlock/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.27/internal/frontend/desktop/darwin/frontend.go:288 +0x2a5
github.com/wailsapp/wails/v2/internal/frontend/desktop/darwin.(*Frontend).startRequestProcessor(0x0)
	/Users/billmatlock/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.27/internal/frontend/desktop/darwin/frontend.go:105 +0x48
created by github.com/wailsapp/wails/v2/internal/frontend/desktop/darwin.NewFrontend
	/Users/billmatlock/go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.27/internal/frontend/desktop/darwin/frontend.go:92 +0x2b1
````
	
	
This also occurs when using the built-in Svelte template installed via: `wails init -n myapp -t svelte -g`

**To Reproduce**
I'm installing this via either of the following methods: 
`wails init -n myapp -t https://github.com/raitonoberu/wails-svelte-template -g`

OR

`wails init -n myapp -t svelte -g`

Then going into `/myapp/frontend` and running:
`npm install`

Then opening a terminal in `/myapp` and running:
`wails dev`

Then in another terminal window in `/myapp/frontend` and running:
`npm run dev`

At this point, running `npm run dev` causes a panic in the first window

**Expected behaviour**
No errors. The Wails app should run and monitor changes in Go files, while the command `npm run dev` watches for changes in web files and rebuilds as necessary.

**System Details**

```
Node: 17.3.0


wails doctor
Wails CLI v2.0.0-beta.27


Scanning system - Please wait (this may take a long time)...Done.

System
------
OS:		MacOS
Version: 	11.6.2
ID:		199506
Go Version:	go1.17.3
Platform:	darwin
Architecture:	amd64

Dependency			Package Name	Status		Version
----------			------------	------		-------
xcode command line tools 	N/A		Installed	2384
npm 				N/A		Installed	8.3.0
*upx 				N/A		Available	

* - Optional Dependency

Diagnosis
---------
Your system is ready for Wails development!
Optional package(s) installation details: 
  - upx : Available at https://upx.github.io/
```

**Additional context**
If I do `npm run dev` first, then in another terminal run `wails dev` everything is fine until I try changing something in a web asset. When Svelte rebuilds, Go panics in the same manner.

**Comment by leaanthony** [comment]
Thanks for the very detailed report! I'm not sure why this would panic in this way but the vanilla template work. I'll investigate when I get chance. Currently moving house 😱

**Comment by Cure4Life** [comment]
So I was finally able to do some debugging on this and here is what I have discovered.

The error seemed to be a result returning from `TranslateUriToFile()` in `go/pkg/mod/github.com/wailsapp/wails/v2@v2.0.0-beta.27/internal/frontend/desktop/common/uri_translate.go` so I added a fmt.Printf line to see what it was getting. It would appear that when live reloading, the URI being sent is `wails://wails:35729/livereload.js?snipver=1`. The url.Host of `wails:35729` is not what is expected so changing the conditional from:

```
if url.Scheme != expectedScheme || url.Host != expectedHost {
	return "", false, nil
}
```

to

```
if url.Scheme != expectedScheme || (url.Host != expectedHost && !strings.Contains(url.Host, expectedHost+":") ) {
	return "", false, nil
}
```

allows it to finish. I am sure this is not the desired fix but it gets me by for now. Hopefully this helps to arrive at a more concrete solution! I have no idea where the `:35729` portion is coming from.

**Comment by leaanthony** [comment]
Thanks for the update! The project doesn't set the host in this way. Where do you think that's coming from? What does your wails.json look like?

**Comment by Cure4Life** [comment]
After some more poking around, apparently this port number is used by Rollup's live reload mechanism. Just doing a text search for `35729` in my project directory, I found reference to it in `frontend/node_modules/rollup-plugin-livereload/index.d.ts`:

```
import { Plugin } from 'rollup'

export interface RollupLivereloadOptions {
  /** Defaults to current directory */
  watch?: string | string[]

  /**
   * Inject the livereload snippet into the bundle which will enable livereload
   * in your web app.
   * Defaults to true 
   */
  inject?: boolean

  /**
   * Log a message to console when livereload is ready
   * Defaults to true 
   */
  verbose?: boolean

  ///
  /// Find all livereload options here:
  /// https://www.npmjs.com/package/livereload#user-content-server-api
  ///

  /**
   * The listening port.
   * It defaults to 35729 which is what the LiveReload extensions use currently.
   */
  port?: number

  /**
   * Add a delay (in milliseconds) between when livereload detects a change to
   * the filesystem and when it notifies the browser. Useful if the browser is
   * reloading/refreshing before a file has been compiled.
   */
  delay?: number

  /** Livereload options, improvements welcome */
  [livereloadOption: string]: any
}

/**
 * 🔄 A Rollup plugin for including livereload in your web app.
 */
export default function livereload(
  options?: RollupLivereloadOptions | string
): Plugin
```

Here is my wails.json for reference:

```
{
  "name": "myapp",
  "frontend:build": "npm run build",
  "frontend:install": "npm install",
  "frontend:dev": "",
  "wailsjsdir": "./frontend",
  "version": "2",
  "Path": "",
  "BuildDir": "",
  "outputfilename": "myapp",
  "OutputType": "",
  "Platform": "",
  "Author": {
    "name": "me",
    "email": "me@example.com"
  },
  "debounceMS": 100,
  "devserverurl": "http://localhost:34115",
  "appargs": ""
}
```

How this is bleeding over into the Wails side of the equation is admittedly outside of my understanding of how these pieces get glued together, but it seems odd to me nevertheless.

**Comment by charlie-dee** [comment]
@BillBuilt

I was toying about with this as I have a related issue. I noticed if you run `wails dev`, inspect element so that the following are loaded:

`DEB | [DesktopAssetServer] Loading file: bundle.js.map`

Make some edit to a frontend file, mine was just App.svelte, save it.

Then run `npm run dev`, it will work... 🤷‍♂️🤔

**Note: Make sure you don't have an existing browser window open on localhost:34115, if you do it will panic.**

**Comment by leaanthony** [comment]
Thanks for digging in more. It's the host name that's confusing the hell out of me. Are you running this in a VM?

**Comment by BillBuilt** [comment]
Nope - natively on a Mac. I originally reached out to the template author and he tested but he's using windows and does not have this happen so it apparently is only an issue on Mac OS but I do not have a windows dev environment set up for testing as of yet so I cannot confirm that. I did create an issue over there though if you want to have a look: https://github.com/raitonoberu/wails-svelte-template/issues/1

**Comment by charlie-dee** [comment]
@leaanthony 

I've been digging and found that when I run `npm run dev` and **TranslateUriToFile** is called, the host is **wails:35729** and not **wails** alone which is what's being checked for.

Hope this helps...

**Comment by charlie-dee** [comment]
A quick fix for this is to remove `:${port} - line 49` from [rollup-plugin-livereload](https://github.com/thgh/rollup-plugin-livereload/blob/master/src/index.js). This also solves my problem in [#1054](https://github.com/wailsapp/wails/discussions/1054#discussion-3772999)

**Comment by charlie-dee** [comment]
Editing the `rollup.config.js` to reflect the below will also resolved the issue. A less tacky fix too 😛

```
!production && livereload({
	watch: 'public',
	inject: false
}),
```

**Comment by charlie-dee** [comment]
Setting `inject: false` in `rollup.config.js` fixes the issue and is less hacky than the before post 😛. However I've had inconsistent behaviour which has me confused. I've had a couple of occasions where livereload didn't work. 

 `wails: v2.0.0-beta.27`

```
!production && livereload({
	watch: 'public',
	inject: false
}),
```

**Comment by BillBuilt** [comment]
I originally wanted to use Vite with Svelte and Wails but could not figure out how to get live reload to work. So that's why I looked to the Svelte template already available. When these issues started popping up I looked to the current Vue + Vite templates to see if there were any clues to getting Svelte + Vite working and I believe I figured out what was causing my initial trials to fail. 

However, I have also had the occasional experience of live reload not working (or better described as it stops working) but it seems that simply closing out both "dev" servers (Wails and Vite) and restarting them kick-starts it into working again. It would also seem that reversing the order of the dev commands _may_ have less issues (frontend first then Wails). 

Although I have not yet had a chance to do any _real_ hardcore development with this, I hope to have the Svelte + Vite template online and available by this weekend. I will continue to monitor this thread and help out with this any way I can though.
